fastlane/spec/actions_specs/register_devices_spec.rb

```diff
@@ -93,6 +93,58 @@
           end").runner.execute(:test)
       end
 
+      describe "registers devices from hash" do
+        it "with name as a key" do
+          expect(Spaceship::ConnectAPI::Device).to receive(:all).and_return(fake_devices)
+
+          devices = {
+            "NAME1" => "A123456789012345678901234567890123456789",
+            "NAME2"	=> "B123456789012345678901234567890123456789"
+          }
+
+          expect(Fastlane::Actions::RegisterDevicesAction).to receive(:try_create_device).with(
+            name: 'NAME2',
+            udid: 'B123456789012345678901234567890123456789',
+            platform: "IOS"
+          )
```

Comment:
[self-review] only 1 request `try_create_device` is expected as `NAME1` already exists from `fake_devices` in line 98.